site/content/blog/2020-12-04-common-i18n-mistakes.md

```diff
@@ -0,0 +1,315 @@
+---
+title: "Avoiding Common Internationalization Mistakes"
+slug: common-i18n-mistakes
+date: 2020-03-26T12:00:00-05:00
+categories:
+    - "i18n"
+author: Harrison Healey
+github: hmhealey
+community: harrison
+---
+
+Languages are complicated, and every language is complicated in different ways that can be hard to understand without learning every single one of them. Some languages form words from multiple characters while others have symbols that represent entire concepts. Some feature words without pluralization or gender and rely on context for that while others have two or even more genders for words. Some are very phonetic while others pronounce words seemingly at random (*cough* English, though *cough*).
+
+Thanks to a tremendous contribution long ago from then community member Elias Nahum, we have full support for translation throughout Mattermost, and thanks to our community of translators, Mattermost is usable by many people speaking a variety of different languages.
+
+Because we support so many languages, we have to be aware of how to keep our applications properly translatable. There's many easy mistakes you can make when writing an application in one language that can make it difficult to translate into others. We use [react-intl](https://formatjs.io/docs/react-intl/) in our client-side applications and [go-i18n](https://github.com/nicksnyder/go-i18n) on the server which both offer a variety of features to help with this, but we can still cause problems if we're not careful when using translated text. Here's a few examples of common problems that we've run into and how to solve them.
+
+## Mistake 1: Not translating something
+
+This one is definitely the easiest to avoid, but it can still catch people who aren't used to writing applications outside their native language.
+
+For example, suppose someone is adding a new button to the web app using React. If they forget to make it to use react-intl, they may write something like this:
```

Comment:
```suggestion
For example, suppose someone is adding a new button to the web app using React. If they forget to make it to use ``react-intl``, they may write something like this:
```

Reply:
Not sure if the formatting suggestion works in this instance, so feel free to ignore if not! I haven't suggested it going forward for that reason.

Reply:
I'll actually rewrite references to that as React Intl since that's how they write it in their docs. For go-i18n, they just refer to it as that, so I don't think it needs to use preformatted text.

Also, I didn't realize you could do inline code with more than one backquote despite all my time working with it 😛